evcache-core/src/main/java/com/netflix/evcache/EVCacheImpl.java

```diff
@@ -215,7 +215,9 @@ EVCacheKey getEVCacheKey(final String key) {
         }
 
         if (canonicalKey.length() > this.maxKeyLength.get() && !hashKey.get() && !autoHashKeys.get()) {
-            throw new IllegalArgumentException("Key is too long (maxlen = " + this.maxKeyLength.get() + ')');
+            final String errMsg = String.format("CanonicalKey ``%s`` is too long (maxLen = %d, keyLen = %d, canonicalKeyLen = %d)", canonicalKey, this.maxKeyLength.get(), key.length(), canonicalKey.length());
```

Comment:
is it possible to add the evcache_app that is causing this? the one that is being written to or read from